src/dune_rules/simple_rules.ml

```diff
@@ -132,9 +132,19 @@ let user_rule sctx ?extra_bindings ~dir ~expander (rule : Rule_conf.t) =
      | Aliases_only aliases ->
        let+ () =
          let action = interpret_and_add_locks ~expander rule.locks action.build in
-         Memo.parallel_iter aliases ~f:(fun alias ->
-           let alias = Alias.make ~dir alias in
-           Alias_rules.add sctx ~alias ~loc:rule.loc action)
+         match List.map ~f:(Alias.make ~dir) aliases with
```

Comment:
Don't we have the same issue above with Aliases_with_targets ?

Reply:
There we are using `Rules.Produce.Alias.add_deps` which correctly registers the deps without introducing any anonymous actions. Here we are using `Super_context.add_alias_action` which is creating an anonymous action. This is a wrapper around `Rules.Produce.Alias.add_action` which for our purposes is probably not what we want.

I've gone ahead and replaced this with `Rules.Produce.Alias.add_deps` here as well and added a test to make sure the with_targets version is treated the same.

Reply:
I can confirm that `with_targets` did not have the issue I am fixing here.